templates/cc/string.go

```diff
@@ -4,64 +4,98 @@ const strTpl = `
 	{{ $f := .Field }}{{ $r := .Rules }}
 	{{ template "const" . }}
 	{{ template "in" . }}
-
 	{{ if $r.Pattern }}
+		{{ unimplemented }}
+		{{/* TODO(akonradi) implement regular expression constraints.
 		if !{{ lookup $f "Pattern" }}.MatchString({{ accessor . }}) {
 			return {{ err . "value does not match regex pattern " (lit $r.GetPattern) }}
 		}
+		*/}}
 	{{ end }}
-
 	{{ if $r.Prefix }}
-		if !strings.HasPrefix({{ accessor . }}, {{ lit $r.GetPrefix }}) {
-			return {{ err . "value does not have prefix " (lit $r.GetPrefix) }}
+	{
+		const std::string prefix = {{ lit $r.GetPrefix }};
+		if ({{ accessor . }}.compare(0, prefix.size(), prefix) != 0) {
```

Comment:
The More You Know: turns out that there is a C++20 and that's when we get starts_with/ends_with in STL..